crates/types/src/config/rocksdb.rs

```diff
@@ -108,8 +110,16 @@ impl RocksDbOptions {
     }
 
     pub fn rocksdb_write_buffer_size(&self) -> NonZeroUsize {
-        self.rocksdb_write_buffer_size
-            .unwrap_or(NonZeroUsize::new(50_000_000).unwrap()) // 50MB
+        // Value is calculated from other defaults of the system
+        self.rocksdb_write_buffer_size.unwrap_or_else(|| {
+            let common_opts = CommonOptions::default();
+            let all_memtables = common_opts.rocksdb_total_memtables_size();
+            let num_partitions = WorkerOptions::default().bootstrap_num_partitions();
```

Comment:
I guess these numbers should not be taken from the defaults but from the actually configured values, right?

Reply:
Those values calculate the default value. I'll add a comment explaining that, but in a possible future, those will be calculated dynamically based on the actual system values (i.e. the current number of configured partitions). We cannot rely on `bootstrap_num_partitions()` because the system can diverge from this value after initial bootstrap.